app/models/miq_worker.rb

```diff
@@ -288,8 +288,9 @@ def self.init_worker_object(*params)
     params[:queue_name]     = default_queue_name unless params.key?(:queue_name) || default_queue_name.nil?
     params[:status]         = STATUS_CREATING
     params[:last_heartbeat] = Time.now.utc
-
-    server_scope.new(params)
+    server_scope.new(params).tap do |w|
+      w.system_uid = w.unit_name if systemd_worker?
```

Comment:
NOTE we have to have an instance in order to access the `unit_name` (specifically because the `guid` is only available after `initialize`) so that is why I have the `.new.tap` here.

TODO I'm sure there is a cleaner way to do this like an after_initialize callback